### ISSUE TITLE
Add new APIs to support read/write file operations for ndjson format

### ISSUE THREAD

**Issue by daneshk** [ballerina-platform/ballerina-standard-library#5099]
**Description:**
The current IO file APIs support read/write JSON files which require complete json object to pass for writing and returns complete object when reading. With ndjson format, which stands for "Newline Delimited JSON,". can store multiple JSON objects in a single file, with each JSON object separated by a newline character. like,

```json
{"name": "John", "age": 30}
{"name": "Alice", "age": 25}
{"name": "Bob", "age": 35}
```

The key features of NDJSON are:
1. Each line in an NDJSON file is a valid JSON object.
2. Lines are separated by newline characters (i.e., "\n").
3. The format is plain text, making it human-readable.

This format is useful specially for storing log data. Each log entry can be represented as a JSON object, and the entries are separated by newlines.

Having new File IO APIs for reading/writing ndjson file is useful for dumping log records to the file and read them later as stream.

**Proposed APIs**

```ballerina
// Read NDJSON file.
public isolated function fileReadNdjson(string path) returns json[]|Error;

// Write NDJSON file.
public isolated function fileWriteNdjson(string path, json[] content, FileWriteOption option = OVERWRITE) returns
Error?;

// Read NDJSON file as stream
public isolated function fileReadNdjsonAsStream(string path) returns stream<json, Error?>|Error;

// Write NDJSON file as stream
public isolated function fileWriteNdjsonFromStream(string path, stream<json, Error?> content,
                                                FileWriteOption option = OVERWRITE) returns Error?;
```

**Related Issues (optional):**
Discord discussion: https://discord.com/channels/957996897782616114/1164531854041427980/1164531854041427980
Resources: https://docs.mulesoft.com/dataweave/2.5/dataweave-formats-ndjson
https://github.com/ndjson/ndjson-spec

**Comment by sameerajayasoma** [comment]
IMO, fileWriteLines is good enough for now. I think we should consider doing this. Should we add this capability to this library? 

@hasithaa, please note this when doing the `data.json` package.

**Comment by daneshk** [comment]
> IMO, fileWriteLines is good enough for now. I think we should consider doing this. Should we add this capability to this library?
> 
Yes, the `fileWriteLines` API works. We may consider improving the `fileWriteLines` API to allow passing JSON objects to support this. For now, it allows only string values.
